MIGRATION.ru.md

```diff
@@ -1,5 +1,778 @@
+
+Методы `findChildBlocks()`, `findParentBlocks()`, `findMixedBlocks()`, `findChildElems()`, `findParentElems()`,
+`findMixedElems()` возвращают [коллекции БЭМ-сущностей](#Коллекции).
+
+Методы `findChildElem()` и `findChildElems()` (в отличие от предыдущего аналога `findElem`) не выполняют поиск на собственных DOM-узлах экземпляра.
+
+Было:
+
+```js
+this.findElem('my-elem');
+```
+
+Стало:
+
+```js
+this.findChildElems('my-elem').concat(this.findMixedElems('my-elem'));
+```
+
+Но рекомендется обратить внимание, действительно ли необходимы оба поиска:
+в большинстве случаев достаточно использовать или `this.findChildElems('my-elem')` или `this.findMixedElems('my-elem')`.
+
+#### Проверка вложенности
+
+Вместо удаленного метода `containsDomElem()`, следует использовать метод `containsEntity()`.
+
+Было:
+
+```js
+this.containsDomElem(someElem);
+```
+
+Стало:
+
+```js
+this.containsEntity(someElem);
+```
+
+### Коллекции
+
+Функциональность элемента `collection` блока `i-bem` перестала быть опциональной.
+
+Все методы возвращавшие массив БЭМ-сущностей, теперь возвращают коллекции.
+
+Было:
+
+```js
+this.findBlocksInside('my-block-2')[0].setMod('my-mod', 'my-val');
+```
+
+Стало:
+
+```js
+this.findBlocksInside('my-block-2').get(0).setMod('my-mod', 'my-val');
+```
+
+Было:
+
+```js
+this.findBlocksInside('my-block-2').forEach(function(myBlock2) {
+    return myBlock2.setMod('my-mod', 'my-val');
+});
+```
+
+Стало:
+
+```js
+this.findChildBlocks('my-block-2').setMod('my-mod', 'my-val');
+```
+
+Задача: [#582](https://github.com/bem/bem-core/issues/582).
+
+### События
+
+API работы с событиями значильно упрощено. Удалены методы экземпляра блока: `on()`, `un()`, `once()`, `bindTo()`,
+`unbindFrom()`, `bindToDoc()`, `bindToWin()`, `unbindFromDoc()`, `unbindFromWin()` и методы класса: `liveBindTo()`,
+`liveUnbindFrom()`, `on()`, `un()`, `once()`, `liveInitOnBlockEvent()`, `liveInitOnBlockInsideEvent()`.
+Вместо них добавлены методы `_domEvents()` и `_events()`, возвращающие экземпляр класса менеджера событий, с методами
+`on()`, `un()` и `one()`;
+
+#### DOM-события на экземплярах
+
+Было:
+
+```js
+BEMDOM.decl('my-block', {
+    onSetMod : {
+        'js' : {
+            'inited' : function() {
+                this.bindTo('click', this._onClick);
+            }
+        }
+    }
+});
+```
+
+Стало:
+
+```js
+bemDom.declBlock('my-block', {
+    onSetMod : {
+        'js' : {
+            'inited' : function() {
+                this._domEvents().on('click', this._onClick);
+            }
+        }
+    }
+});
+```
+
+Было:
+
+```js
+BEMDOM.decl('my-block', {
+    onSetMod : {
+        'js' : {
+            'inited' : function() {
+                this.bindToDoc('click', this._onDocClick);
+            }
+        }
+    }
+});
+```
+
+Стало:
+
+```js
+bemDom.declBlock('my-block', {
+    onSetMod : {
+        'js' : {
+            'inited' : function() {
+                this._domEvents(bemDom.doc).on('click', this._onDocClick);
```

Comment:
Please add info about context of the callback function.

I found out that for
```js
this._domEvents(bemDom.doc).on('pointerrelease', this._onPointerRelease);
```
`this._onPointerRelease` function will be called with `document` as a context.
I expected that I can provide context with additional optional argument like this:
```js
this._domEvents(bemDom.doc).on('pointerrelease', this._onPointerRelease, this);
```
But it didn't help. And the only way I found is
```js
this._domEvents(bemDom.doc).on('pointerrelease', this._onPointerRelease.bind(this));
```

Is it expected behaviour or a bug?